test/suite/test_txn19.py

```diff
@@ -0,0 +1,331 @@
+            log.truncate()
+        if offset:
+            if offset < 0:  # Negative offset means seek to the end
+                log.seek(0, 2)
+            else:
+                log.seek(offset)
+        if writeit:
+            log.write(writeit)
+
+class test_txn19(wttest.WiredTigerTestCase, suite_subprocess):
+    base_config = 'log=(archive=false,enabled,file_max=100K),' + \
+                  'transaction_sync=(enabled,method=none)'
+    conn_config = base_config
+    corruption_type = [
+        ('removal', dict(kind='removal', f=lambda fname:
+            os.remove(fname))),
+        ('truncate', dict(kind='truncate', f=lambda fname:
+            corrupt(fname, True, 0, None))),
+        ('zero-begin', dict(kind='zero', f=lambda fname:
+            corrupt(fname, False, 0, '\0' * 4096))),
+        ('zero-trunc', dict(kind='zero', f=lambda fname:
+            corrupt(fname, True, 0, '\0' * 4096))),
+        ('zero-end', dict(kind='zero-end', f=lambda fname:
+            corrupt(fname, False, -1, '\0' * 4096))),
+        ('garbage-begin', dict(kind='garbage-begin', f=lambda fname:
+            corrupt(fname, False, 0, 'Bad!' * 1024))),
+        ('garbage-middle', dict(kind='garbage-middle', f=lambda fname:
+            corrupt(fname, False, 1024 * 25, 'Bad!' * 1024))),
+        ('garbage-end', dict(kind='garbage-end', f=lambda fname:
+            corrupt(fname, False, -1, 'Bad!' * 1024))),
+    ]
+    # The list comprehension below expands each entry in the integer tuple
+    # list to a scenario.  For example, (3, 4, 2) expands to:
+    # ('corrupt=[3,4],checkpoint=2', dict(corruptpos=3, corruptpos2=4, chkpt=2))
+    #
+    # Each number corresponds to a log file, so for this example, we have
+    # corruption in log file 3 (using the style of corruption from
+    # corruption_type), there is a second corruption in log file 4,
+    # and there is a checkpoint in log file 2.  A value of 0 means no
+    # corruption or checkpoint.
+    corruption_pos = [
+        ('corrupt=[' + str(x) + ',' + str(y) + '],checkpoint=' + str(z),
+           dict(corruptpos=x,corruptpos2=y,chkpt=z)) for (x,y,z) in (
+               (0, 0, 0), (0, 0, 2), (6, 0, 0), (6, 0, 3), (3, 0, 0),
+               (3, 0, 2), (3, 4, 2), (3, 5, 2), (3, 0, 4))]
+    nrecords = [('nrecords=10', dict(nrecords=10)),
+                ('nrecords=11', dict(nrecords=11))]
+
+    # This function prunes out unnecessary or problematic test cases
+    # from the list of scenarios.
+    def includeFunc(name, dictarg):
+        kind = dictarg['kind']
+        corruptpos = dictarg['corruptpos']
+        chkpt = dictarg['chkpt']
+
+        # corruptpos == 0 indicates there is no corruption.
+        # (i.e. corrupt log file 0, which doesn't exist)
+        # We do want to test the case of no corruption, but we don't
+        # need to try it against every type of corruption, only one.
+        if corruptpos == 0:
+            return kind == 'removal'
+
+        # XXX
```

Comment:
Is this something you meant to get back to? Or is this for a later ticket? Maybe change XXX to NOTE:

Reply:
@sueloverso, I opened three tickets WT-4147, WT-4148, WT-4149 to cover these XXX spots in the tests.  I changed comments to NOTE: to be more palatable in the meantime.
Thanks for the review!